lib/vehicle_factory.rb

```diff
@@ -0,0 +1,20 @@
+class VehicleFactory
+  attr_reader :new_vehicles
+
+  def initialize
+    @new_vehicles = []
+  end
+
+  def create_vehicles(state_registrations)
+    state_registrations.select do |vehicle|
+      model = vehicle[:model]
+      vin = vehicle[:vin_1_10]
+      year = vehicle[:model_year]
+      make = vehicle[:make]
+
+      new_vehicle = Vehicle.new({vin: vin, year: year, make: make, model: model, engine: :ev})
+      @new_vehicles << new_vehicle
+    end
+    @new_vehicles
+  end
```

Comment:
If you used a `map` rather than `select`, you wouldn't need to use a placeholder to accumulate the vehicles that you create. In fact, the way you have it set up right now, `each` would be a better option than `select`, because you're iterating through a collection, doing something, and then returning the array you used to accumulate. If you use `map`, and change each vehicle data into a vehicle object, then you have your array already for you! 

```ruby
  def create_vehicles(registrations)
     registrants.map do |vehicle|
        Vehicle.new({
          engine: :ev,
          make: vehicle[:make],
          model: vehicle[:model],
          vin: vehicle[:vin_1_10],
          year: vehicle[:model_year]
        })
    end 
 end
```